src/core/public/chrome/nav_links/nav_links_service.ts

```diff
@@ -95,39 +95,34 @@ export interface ChromeNavLinks {
   getForceAppSwitcherNavigation$(): Observable<boolean>;
 }
 
+type LinksUpdater = (navLinks: Map<string, NavLinkWrapper>) => Map<string, NavLinkWrapper>;
+
 export class NavLinksService {
   private readonly stop$ = new ReplaySubject(1);
 
   public start({ application, http }: StartDeps): ChromeNavLinks {
-    const appLinks = [...application.availableApps]
-      .filter(([, app]) => !app.chromeless)
-      .map(
-        ([appId, app]) =>
-          [
-            appId,
-            new NavLinkWrapper({
-              ...app,
-              legacy: false,
-              baseUrl: relativeToAbsolute(http.basePath.prepend(`/app/${appId}`)),
-            }),
-          ] as [string, NavLinkWrapper]
-      );
-
-    const legacyAppLinks = [...application.availableLegacyApps].map(
-      ([appId, app]) =>
-        [
-          appId,
-          new NavLinkWrapper({
-            ...app,
-            legacy: true,
-            baseUrl: relativeToAbsolute(http.basePath.prepend(app.appUrl)),
-          }),
-        ] as [string, NavLinkWrapper]
+    const appLinks$ = application.availableApps$.pipe(
+      map(apps => {
+        return new Map(
+          [...apps]
+            .filter(([, app]) => !app.chromeless)
+            .map(([appId, app]) => [appId, toNavLink(app, http.basePath)])
+        );
+      })
     );
+    const linkUpdaters$ = new BehaviorSubject<LinksUpdater[]>([]);
```

Comment:
A comment explaining this would useful. Though I know from our conversations, I don't think it'll be obvious to us in the future why we need these functions.

Reply:
Futur me definitely agrees with you